### ISSUE TITLE
distance controlling between vehicles[FEATURE]

### ISSUE THREAD

**Issue by LukasDO665** [klyte45/TransportLinesManager#206]
Hey!
I think the mod is really good, but one feature would be very nice!
In my city, it is not uncommon for several buses on one route to queue up at one stop. I definitely don't have too many buses on one line. It's a game bug. The Improved Public Transport Mod fixes this problem. However, I like this mod more. So I would be very happy if this problem would be fixed with an update. Then it would be perfect! :)

**Comment by draco963** [comment]
I really hope this gets implemented soon. With IPT2 having been broken by the Airport DLC I've been trying to use this, but all my buses just bunch up at the first stop on their line.

**Comment by Xytronix** [comment]
> Hey! I think the mod is really good, but one feature would be very nice! In my city, it is not uncommon for several buses on one route to queue up at one stop. I definitely don't have too many buses on one line. It's a game bug. The Improved Public Transport Mod fixes this problem. However, I like this mod more. So I would be very happy if this problem would be fixed with an update. Then it would be perfect! :)

What is IPT2 doing instead?

**Comment by LukasDO665** [comment]
> > Hey! I think the mod is really good, but one feature would be very nice! In my city, it is not uncommon for several buses on one route to queue up at one stop. I definitely don't have too many buses on one line. It's a game bug. The Improved Public Transport Mod fixes this problem. However, I like this mod more. So I would be very happy if this problem would be fixed with an update. Then it would be perfect! :)
> 
> What is IPT2 doing instead?

The IPT2 mod controlls the distance between two vehicles. If they have a distance where they could queue up on a station, they are waiting instead until the distance is reached, where they can drive without queuing up. Sorry for my bad english :D

**Comment by draco963** [comment]
As I understand it, IPT2 makes each bus check the distance between itself
and the bus in front of it. It (allegedly) takes the distance of the line
as the dividend [D] , uses the number of buses (vehicles, actually, because
this math is performed on every transit route, not just buses) as the
divisor [V] , and then forces each vehicle to wait until the distance to
the vehicle in front of it is at least equal to the quotient [Q] (D/V=Q).

So, on a 4km bus-route with 10 buses on it, theoretically each bus would
wait, at each stop, for the bus ahead of it to be 400m away before
departing whichever stop it was at (4000/10=400).

I want to be clear, this is an *understanding* that I have gained from
reading the comments and watching the gameplay. While I have modded other
games (some extensively), I have not done any modding for C:S, and I have
not examined the code of BloodyPenguin's IPT2.

Given that caveat, I suspect I'm missing something from the math proposed
above, because forcing each vehicle to rigidly keep to Q would, I believe,
cause a continues slowing progressing backwards through the route, as the
response time for vehicles is quite slow. I suspect the Q value must take
that into account *somehow*, perhaps the formula is Q=D/V+1, or
Q=(D×0.9)/V, *something* to give the vehicles that wee bit of time they
need to actually pull away from the stop. But I haven't actually tried to
open up IPT2 to try and find it.

I hope this helps. Thank you Klyte. I really appreciate your work, your
temp LSM mod got me through this past week!

**Comment by Jaroch76** [comment]
I agree.  While TLM is far more powerful than IPT2, I really miss IPT2's "unbunching" feature because it often feels like my busses aren't well spaced out when using TLM.

**Comment by jamesq63** [comment]
I don't really notice this issue unless I have "express bus services" enabled. Have you checked to see if this is enabled in the options?
My understanding is that Cities Skylines spaces vehicles "unbunching" by default.

**Comment by Jaroch76** [comment]
James:

Good point about the express busses.  And thank you, by the way, for responding.

First, recent reading has shown that my last comment is outdated.  I was thinking back to the days when unbunching wasn't a part of vanilla _Cities_.  That's what I get for not paying more attention to the literature... what little there is.

I've also never thought to disable TLM's express bus feature to see how that affected things.  After all, the express buses were on3 of the big selling points of TLM.  I did so now and it helped somewhat.  In retrospect, I seems kind of obvious that express busses and unbunched busses are mutually contradictory.

Based on this test, I'd say that it'd still be nice to have the option to tweak the vanilla unbunching settings.  But, it'd also be nice to have express buses be line-specific instead of all-or-nothing.

**Comment by praveenjabakar** [comment]
Dear Developer,

May be you can use this. I use this on top of TLM. Works perfectly ( just divide method gap , no of vehicles and distance i think) . I am not a fan of c#. Please check it for an idea. May be you can add this with enable and disable button for each seperate line. Because this vehicle unbouncer applies to all, not specific to line or to specifc type.

https://steamcommunity.com/sharedfiles/filedetails/?id=531401164&searchtext=vehicle+unbouncer

Source code
You can download it here: https://cld.pt/dl/download/80875646-ff11-4853-b142-0ba3092a35ca/VehicleUnbuncher.zip

Thanks for your support and help

**Comment by jamesq63** [comment]
> I've also never thought to disable TLM's express bus feature to see how that affected things. After all, the express buses were on3 of the big selling points of TLM. I did so now and it helped somewhat. In retrospect, I seems kind of obvious that express busses and unbunched busses are mutually contradictory.

I agree...I was excited for express buses but cant use it because of bunching...Some sort of timetable feature would be nice - using Real Time - bus waits until 10:00am to depart from terminal stop etc.

**Comment by Jaroch76** [comment]
> Dear Developer,
> 
> May be you can use this. I use this on top of TLM. Works perfectly ( just divide method gap , no of vehicles and distance i think) . I am not a fan of c#. Please check it for an idea. May be you can add this with enable and disable button for each seperate line. Because this vehicle unbouncer applies to all, not specific to line or to specifc type.
> 
> https://steamcommunity.com/sharedfiles/filedetails/?id=531401164&searchtext=vehicle+unbouncer
> 
> Source code You can download it here: https://cld.pt/dl/download/80875646-ff11-4853-b142-0ba3092a35ca/VehicleUnbuncher.zip
> 
> Thanks for your support and help

Thank you for the link.  I've tested it out before, and it didn't seem to make much difference.  But, then, it was probably fighting against TLM's express-bus feature.  I'll keep it in mind for the future, but I suspect that I'm just going to turn off express buses, as they're not that useful in my city.

**Comment by praveenjabakar** [comment]
But with express bus enabled and with this vehicle unbouncer mod, like express bus didnt stop in the stop but next/previous bus in the starting terminal or in another bus stop, it waits for distance calculation and resumes. Yes first time, all vechicle waits at starting terminal like 30 busses in the starting terminal but when games moves on, it was flowing smoothly. Anyway you know better. Thanks brother.

**Comment by klyte45** [comment]
Okay guys, I guess I got the point!

I will try to find a way to control the unbunching when using express lines. I understand the stop times in the stops will be shorter when using this feature, then the vanilla value is causing this sensation of too much buses stopped at terminal stops.
